File: store-core/control-plane/control-plane-service/src/main/java/com/asrevo/cvhome/controlplane/manager/service/SignupService.java
package com.asrevo.cvhome.controlplane.manager.service;

import com.asrevo.cvhome.controlplane.manager.dto.CreateOrgRequest;
import com.asrevo.cvhome.uaa.api.errors.UaaApiUnavailableException;
import com.asrevo.cvhome.uaa.api.errors.UaaConflictException;
import com.asrevo.cvhome.uaa.domain.user.ReadableUser;

public interface SignupService {

    /**
     * Creates the organization and its first user.
     *
     * @throws UaaConflictException        that email is already registered in uaa, so the organization must not be
     *                                     presented as created
     * @throws UaaApiUnavailableException  uaa could not be reached; the organization row may already exist, so this
     *                                     needs reconciling rather than retrying blindly
     */
    ReadableUser createOrgUser(CreateOrgRequest request) throws UaaConflictException, UaaApiUnavailableException;

}

File: store-core/console-ui/src/app/api/signup/sign-up.service.ts
import {Injectable, inject} from '@angular/core';
import {Observable} from 'rxjs';

import {CrudService} from '@cvhome-saas/ui-kit';
import type {CreateOrgRequest, ReadableUser} from '@models/signup';

export const SIGNUP_API_BASE = '/tenancy/api/v1/signup/public';

@Injectable({providedIn: 'root'})
export class SignUpService {
  private readonly crudService = inject(CrudService);

  /**
   * Creates the organization and its first administrator, or neither — the server holds both in one transaction.
   *
   * Conflicts (`UaaConflictException`, e.g. the email is already registered) arrive as an RFC-7807 problem with a
   * `fieldErrors[]` entry, so callers should hand the error to `ApiErrorService.applyToForm` rather than toasting it.
   */
  signUp(request: CreateOrgRequest): Observable<ReadableUser> {
    return this.crudService.post(`${SIGNUP_API_BASE}/create`, request);
  }
}
